src/rt/sections_elf_shared.d

```diff
@@ -809,6 +872,12 @@ bool findDSOInfoForAddr(in void* addr, dl_phdr_info* result=null) nothrow @nogc
          */
         return dl_iterate_phdr(&callback, &dg) != 0;
     }
+    else version (OSX)
+    {
+        auto header = _dyld_get_image_header_containing_address(addr);
+        if (result) *result = header;
+        return !!header;
```

Comment:
Without knowing this existed, I independently came up with this implementation for findDSOInfoForAddr:
```
foreach (i; 0 .. _dyld_image_count())
{
    const header = _dyld_get_image_header(i);
    const slide = _dyld_get_image_vmaddr_slide(i);
    const section = getSection(header, slide, SEG_DATA, SECT_DATA);

    // less than base address of section means quick reject
    if (!section.length || addr < section.ptr)
       continue;

    if (addr < section.ptr + section.length)
    {
        result.header = header;
        result.slide = slide;
        return true;
    }
}
return false;
```
Probably could align ourselves up, mixing what you have here with `_dyld_get_image_vmaddr_slide` for getting the slide value.  So you don't have to rely on a private function later.

Reply:
On darwin9, and older versions of darwin10, will have to use `dladdr()` instead.

Reply:
This PR is more than 2 years old - this now looks like this: https://github.com/ldc-developers/druntime/blob/a261a84e4c6b208c65ae39da0ae844ae3dd06744/src/rt/sections_elf_shared.d#L903-L907

Reply:
The slide can be calculated like this [1]. Also note this [2]:

"The following functions allow you to iterate through all loaded images. This is not a thread safe operation. Another thread can add or remove an image during the iteration.  
 
Many uses of these routines can be replace by a call to dladdr() which  will return the mach_header and name of an image, given an address in  the image. dladdr() is thread safe."

[1] https://github.com/ldc-developers/druntime/blob/a261a84e4c6b208c65ae39da0ae844ae3dd06744/src/rt/sections_darwin_64.d#L134

[2] https://opensource.apple.com/source/dyld/dyld-750.6/include/mach-o/dyld.h.auto.html

Reply:
@jacob-carlborg good to know, though you're still using [_dyld_image_count and get_image_header](https://github.com/ldc-developers/druntime/blob/a261a84e4c6b208c65ae39da0ae844ae3dd06744/src/rt/sections_darwin_64.d#L40-L42)

Reply:
Yeah, IIRC I couldn't get that to work with `dladdr`.